lib/counter.js

```diff
@@ -388,4 +388,67 @@ TimestampedCounter.prototype._countRange = function(keys, eventObj, cb) {
   }
 };
 
+/**
+ * Parse a rank result from redis
+ * @param  {array} rank In this format: [ 'foo', '39', 'bar', '13' ]
+ * @return {object} In this format: { foo: 39, bar: 13 }
+ */
+var rankParser = function(rank) {
+  rank = _.groupBy(rank, function(a, b) {
+    return Math.floor(b / 2);
+  });
```

Comment:
Basically a is the array entry and b is the index. `groupBy` will group the array entries in an object with keys based on the callback result, so in this case from `[ 'foo', '39', 'bar', '13' ]` will be translated to `{ '0': [ 'foo', '39' ], '1': [ 'bar', '13' ] }`